sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppImpl.java

```diff
@@ -371,45 +372,47 @@ public Mono<KuduDeploymentResult> pushDeployAsync(DeployType type, File file, De
     }
 
     @Override
-    public DeploymentBuildStatus getDeploymentStatus(String deploymentId) {
+    public CsmDeploymentStatus getDeploymentStatus(String deploymentId) {
         return getDeploymentStatusAsync(deploymentId).block();
     }
 
     @Override
-    public Mono<DeploymentBuildStatus> getDeploymentStatusAsync(String deploymentId) {
+    public Mono<CsmDeploymentStatus> getDeploymentStatusAsync(String deploymentId) {
         // "GET" LRO is not supported in azure-core
-        String deploymentStatusUrl = String.format(
-            "%s/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Web/sites/%s/deploymentStatus/%s?api-version=%s",
-            AzureEnvironment.AZURE.getResourceManagerEndpoint(),
-            this.manager().subscriptionId(),
-            this.resourceGroupName(),
-            this.name(),
-            deploymentId,
-            this.manager().serviceClient().getApiVersion());
-        HttpRequest request = new HttpRequest(HttpMethod.GET, deploymentStatusUrl);
-        return this.manager().httpPipeline().send(request)
-            .flatMap(response -> {
-                if (response.getStatusCode() / 100 != 2) {
-                    return Mono.error(new ManagementException("Service responds with a non-20x response.", response));
+        AtomicLong pollCount = new AtomicLong();
+        Duration pollDuration = manager().serviceClient().getDefaultPollInterval();
+        return this.manager().serviceClient().getWebApps()
+            .getProductionSiteDeploymentStatusWithResponseAsync(this.resourceGroupName(), this.name(), deploymentId)
+            .flatMap(fluxResponse -> {
+                if (pollDuration.multipliedBy(pollCount.get()).compareTo(MAX_DEPLOYMENT_STATUS_TIMEOUT) < 0) {
+                    HttpResponse response = new HttpFluxBBResponse(fluxResponse);
+                    if (fluxResponse.getStatusCode() / 100 != 2) {
+                        return Mono.error(new ManagementException("Service responds with a non-20x response.", response));
+                    }
+                    return response.getBodyAsString()
+                        .flatMap(bodyString -> {
+                            SerializerAdapter serializerAdapter = JacksonAdapter.createDefaultSerializerAdapter();
```

Comment:
mgmt uses this https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core-management/src/main/java/com/azure/core/management/serializer/SerializerFactory.java